### PR TITLE
doc/user: use newly open-sourced diagram generator

### PR DESCRIPTION
**Pull request by benesch**
(This is very low priority, but it's been a skunkworks project of mine for a while, and I needed a break from the Kafka stuff this evening.)

The railroad diagram generator we use has been [newly open-sourced](https://github.com/GuntherRademacher/rr), which means we can now customize it to our liking without needing to post-process the generated SVGs.

@sploiselle the big exciting feature here is that we can swap in our own font width tables and render the diagrams in the marketing font du jour, rather than Verdana. I'm sad to see the old Go diagram generator go, so definitely let me know if you have concerns re: modifying the Java generator, or any SVG modifications we were doing before but no longer.

Ultimately I'd like to avoid checking in the SVGs at all, and generate them on the fly when someone runs `hugo serve`, but that's a project for another day.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/3075)

### CODE REVIEW

File: doc/user/README.md [hunk 1]
```diff
@@ -62,26 +60,24 @@ Railroad diagrams consist of two parts:
...
    <https://www.bottlecaps.de/rr/ui>.
-2. Go to `util/rr-diagram-gen/`
-3. Run `make run`.
+2. Run `sql-grammar/generate.sh`.
 
 You can now include the diagram on a page using:
 
 ```html
-{{&lt; diagram "SOME_DIAGRAM.html" &gt;}}
+{{< diagram "SOME-NONTERMINAL.svg" >}}
```
Comment (sploiselle) [inline doc/user/README.md:77]: Man; I tried for like 15 minutes one day to get this to work and it just absolutely refused. Does the markdown actually render this as a code block?

Comment (benesch) [reply doc/user/README.md:77]: I think so! https://github.com/MaterializeInc/materialize/blob/4023f7d6ce771c8961fdc8b976c47436d98cbcaf/doc/user/README.md#to-generate-diagrams

Wanna double check that it works for you too? Maybe it's a browser/platform thing?

Comment (sploiselle) [reply doc/user/README.md:77]: I'll be damned. Works for me. Sorry I didn't actually look!

Comment (benesch) [reply doc/user/README.md:77]: No no, glad it was that easy!

**Pull request merged**